src/features/transactions/Transactions.tsx

```diff
@@ -1,13 +1,47 @@
-import React from "react";
+import { useEffect } from "react";
 import { getEtherscanURL } from "@airswap/utils";
 import { Web3Provider } from "@ethersproject/providers";
 import { useWeb3React } from "@web3-react/core";
-import { useAppSelector } from "../../app/hooks";
-import { selectTransactions } from "./transactionsSlice";
+import { useAppSelector, useAppDispatch } from "../../app/hooks";
+import { selectTransactions, setTransaction } from "./transactionsSlice";
+import {
+  revertTransaction,
+  mineTransaction,
+} from "./transactionActions";
+import { getTransactionsLocalStorageKey } from "../metadata/metadataApi";
 
 export function Transactions() {
   const transactions = useAppSelector(selectTransactions);
-  const { active, chainId } = useWeb3React<Web3Provider>();
+  const dispatch = useAppDispatch();
+  const { active, chainId, library } = useWeb3React<Web3Provider>();
+
+  useEffect(() => {
+    // get transaction state from local storage and update the transactions
+    if (chainId) {
+      const transactionsLocalStorage = JSON.parse(localStorage.getItem(getTransactionsLocalStorageKey(chainId!))!);
+
+      dispatch(setTransaction(transactionsLocalStorage))
+
+      // check from all responses if one is pending... if pending, call getTransaction
+      // to see if it was a success/failure/pending. update accordingly. if pending: wait()
+      // and poll at a sensible interval.
+      // TODO: Poll at sensible intervals
+      transactions.forEach(tx => {
```

Comment:
On second look at this, I don't think you should be using `transactions` here - instead you want to use `transactionsLocalStorage`. I have a feeling `transactions` will actually not be updated at this point anyway (`transactions` would be computed on line 14 when rendering, and will not have updated between 23 and 29), but the hint is that we actually don't want this `useEffect` to have a dependency on `transactions`, which it should if it uses transactions. In other words, this effect doesn't care about the redux state of transactions changing, it only cares about transactions that were pending in localstorage at the time it ran.

Two final notes:

- Our local storage keys need to be updated to have the account in them as well as the chainId, so if I switch account I only see transactions (and activeTokens) for the account I've switched to.
- When we switch chains or accounts, we really need to cancel listeners and polling for all transactions. This effect will run to set all of those back up again if we switch back. <- this one is complex, maybe leave it 'til last and we can run through this together.